src/iscan.cpp

```diff
@@ -100,23 +100,49 @@ usage() {
+        "A full physical path must be used for local files and directories.",
         " ",
-        "Full path must be used for local files and directories.",
-        " ",
-        "If the input is an iRODS file or an iRODS collection, iscan",
-        "checks if the physical files corresponding to the iRODS object",
-        "exists on the data servers. Scanning data objects and",
-        "collections may only be performed by a rodsadmin.",
+        "If the input is a logical path representing an iRODS data object",
+        "or an iRODS collection, iscan checks if the associated registered",
+        "physical file entries exist on the data servers.",
+        "Scanning data objects and collections may only be performed by a rodsadmin.",
         " ",
         "If the operation is successful, nothing will be output",
-        "   and 0 will be returned.",
+        "and 0 will be returned.",
         " ",
         "Options are:",
-        " -r  recursive - scan local subdirectories or subcollections",
+        " -r  recursive - scan local subdirectories or logical subcollections",
         " -h  this help",
-        " -d  scan data objects in iRODS (default is scan local objects)",
+        " -d  scan data objects in iRODS (default is scan local files/directories)",
+        " ",
+        "Examples:",
+        "# scan a single physical path to check its presence in the catalog",
+        "$ iscan /var/lib/irods/not_in_catalog",
+        "/var/lib/irods/not_in_catalog is not registered in iRODS",
+        " ",
+        "# recursively scan a physical path to check for unregistered local files",
+        "$ iscan -r /var/lib/irods/Vault/home/rods/scanned",
+        "/var/lib/irods/Vault/home/rods/scanned/does_not_exist2 is not registered in iRODS",
+        "/var/lib/irods/Vault/home/rods/scanned/does_not_exist1 is not registered in iRODS",
+        " ",
+        "# recursively scan a physical path to check for unregistered local files",
+        "$ iscan -r /data/projects/myproject",
+        "remote addresses: 127.0.0.1 ERROR: scanObj: /data/projects/myproject does not exist",
+        " ",
+        "# check that a single relative logical path exists in the catalog",
+        "$ iscan -d not_a_dataobject",
+        "Could not find the requested data object or collection in iRODS.",
+        " ",
+        "# recursively scan a relative logical path for missing physical paths",
+        "$ iscan -rd scanme",
+        "Physical file /var/lib/irods/Vault/home/rods/scanme/does_not_exist on server irods.example.org is missing, corresponding to iRODS object /tempZone/home/rods/scanme/does_not_exist",
+        "Physical file /var/lib/irods/Vault/home/rods/scanme/missing on server irods.example.org is missing, corresponding to iRODS object /tempZone/home/rods/scanme/missing",
+        " ",
+        "# rodsadmin required to scan logical paths",
+        "$ iscan -d test.txt",
+        "User must be a rodsadmin to scan iRODS data objects.",
```

Comment:
Does this mean `iscan` cannot be used by rodsusers/groupadmins?

Reply:
nonadmins can scan physical paths, but not logical paths

Reply:
Got it.